src/sumlog.jl

```diff
@@ -0,0 +1,36 @@
+using IrrationalConstants: logtwo
+
+"""
+$(SIGNATURES)
+
+Compute `sum(log.(X))`.
+
+`sum(log.(X))` can be evaluated much more quickly as `sum(log, X)`. However,
+this still requires computing `log` for each element of `X`. 
+
+`sumlog(X)` can be faster still, especially an the length of `X` increases.
+
+This works by representing the `j`th element of `X` as `xⱼ = aⱼ * 2 ^ bⱼ`,
+allowing us to write
+
+    ∑ⱼ log(xⱼ) = log(∏ⱼ aⱼ) + log(2) * ∑ⱼ bⱼ
+
+Since `log(2)` is constant, `sumlog` only requires a single `log` evaluation.
+"""
+function sumlog(x::AbstractArray{T}) where {T} 
```

Comment:
It seems this will fail for non-floating point types `T` such as `Int`, etc., `BigFloat`, and for complex numbers?